torch/tensor.py

```diff
@@ -426,6 +427,19 @@ def __array_wrap__(self, array):
             array = array.astype('uint8')
         return torch.from_numpy(array)
 
+    def __contains__(self, element):
+        r"""Check if `element` is present in tensor
+
+        Arguments:
+            element (Tensor or scalar): element to be checked
+                for presence in current tensor"
+        """
+        if torch.is_tensor(element) or isinstance(element, Number):
+            return (element == self).any().item()
+        raise NotImplementedError(
```

Comment:
hmm I think you should return `NotImplemented` rather than raising `NotImplementedError` so that the default behavior using `__iter__` can be invoked.

Reply:
strainge observation
```
>>> a = torch.arange(10)
>>> 'b' in a   // this returns True, because of fall back to __iter__
```

Reply:
string `'b'`? that doesn't look right...

Reply:
Yeah.
I checked about NotImplemented.
Looks like we are typecasting it to bool and returning somewhere
`bool(NotImplemented) is true`
That's why it's not falling back to `__iter__` based method

btw, why should we fall back to basic method as it will still lead to error if it's not tensor or scalar

And Isn't specific error by us preferable than relying on standard `__iter__` based method to give random error.

ref: https://medium.com/@s16h/pythons-notimplemented-type-2d720137bf41